sdk/core/core-https/src/pipeline.ts

```diff
@@ -343,3 +365,100 @@ class HttpsPipeline implements Pipeline {
 export function createEmptyPipeline(): Pipeline {
   return HttpsPipeline.create();
 }
+
+/**
+ * Options that allow configuring redirect behavior.
+ */
+export interface PipelineRedirectOptions extends RedirectPolicyOptions {
+  /**
+   * If true, disables automatic following of redirects.
+   */
+  disable?: boolean;
+}
+
+/**
+ * Defines options that are used to configure the HTTP pipeline for
+ * an SDK client.
+ */
+export interface PipelineOptions {
+  /**
+   * The HttpsClient implementation to use for outgoing HTTP requests.
+   * Defaults to DefaultHttpsClient.
+   */
+  httpClient?: HttpsClient;
+
+  /**
+   * Options that control how to retry failed requests.
+   */
+  retryOptions?: ExponentialRetryPolicyOptions;
+
+  /**
+   * Options to configure a proxy for outgoing requests.
+   */
+  proxyOptions?: ProxySettings;
+
+  /**
+   * Options for how HTTP connections should be maintained for future
+   * requests.
+   */
+  keepAliveOptions?: KeepAlivePolicyOptions;
+
+  /**
+   * Options for how redirect responses are handled.
+   */
+  redirectOptions?: PipelineRedirectOptions;
+
+  /**
+   * Options for adding user agent details to outgoing requests.
+   */
+  userAgentOptions?: UserAgentPolicyOptions;
+}
+
+/**
+ * Defines options that are used to configure internal options of
+ * the HTTP pipeline for an SDK client.
+ */
+export interface InternalPipelineOptions extends PipelineOptions {
+  /**
+   * Options to configure request/response logging.
+   */
+  loggingOptions?: LogPolicyOptions;
+
+  /**
+   * Configure whether to decompress response according to Accept-Encoding header (node-fetch only)
+   */
+  decompressResponse?: boolean;
+}
+
+/**
+ * Create a new pipeline with a default set of customizable policies.
+ * @param options Options to configure a custom pipeline.
+ */
+export function createPipelineFromOptions(options: InternalPipelineOptions): Pipeline {
+  const pipeline = HttpsPipeline.create();
+
+  if (isNode) {
+    pipeline.addPolicy(proxyPolicy(options.proxyOptions));
+
+    if (options.decompressResponse === false) {
+      pipeline.addPolicy(disableResponseDecompressionPolicy());
+    }
+  }
+
+  pipeline.addPolicy(formDataPolicy());
+  pipeline.addPolicy(tracingPolicy(options.userAgentOptions));
+  pipeline.addPolicy(keepAlivePolicy(options.keepAliveOptions));
+  pipeline.addPolicy(userAgentPolicy(options.userAgentOptions));
+  pipeline.addPolicy(setClientRequestIdPolicy());
+  pipeline.addPolicy(throttlingRetryPolicy(), { phase: "Retry" });
+  pipeline.addPolicy(systemErrorRetryPolicy(options.retryOptions), { phase: "Retry" });
```

Comment:
Doesn't have to be changed now since this matches current state, but technically we don't need the `systemErrorRetryPolicy` in browsers do we?

Reply:
Can't XHR still face transient errors like this? Maybe it doesn't show up like this, but I remember it has some kind of error for if you get knocked offline

Reply:
I don't think the `code`s used in this policy match up with those that can be assigned to DOMExceptions:
https://heycam.github.io/webidl/#idl-DOMException
https://heycam.github.io/webidl/#idl-DOMException-error-names

That said, I don't know that for sure, so probably better to leave it in for now until if/when we do some testing to see.

Reply:
Yeah, I suspect our browser support for this policy is woefully deficient. I filed #9568 to follow-up on this later.